core/src/smartcontracts/wasm.rs

```diff
@@ -125,10 +132,19 @@ impl<'a, W: WorldTrait> Runtime<'a, W> {
     /// Host defined function which executes ISI. When calling this function, module
     /// serializes ISI to linear memory and provides offset and length as parameters
     ///
+    /// # Warning
+    ///
+    /// This function doesn't take ownership of the provided allocation
```

Comment:
the return type of this function is just `()`. `Result<(), Trap>` is just `wasmtime`'s sugar on top of it. For `()` it doesn't matter that ownership is transferred to the caller. I left this comment in case we return something else here

But you are correct, `#[must_use]` is very adequate both here and for `execute_query`. Nice catch